pkg/operation/botanist/kubeapiserverexposure.go

```diff
@@ -151,3 +151,32 @@ func (b *Botanist) setAPIServerServiceClusterIP(clusterIP string) {
 		},
 	)
 }
+
+// DeployKubeAPIServerIngress deploys ingress for the kube-apiserver.
+func (b *Botanist) DeployKubeAPIServerIngress(ctx context.Context) error {
```

Comment:
Unit tests for these functions might be useful to make sure that the `ingressClass` is specified proprely and to make sure that when `b.ControlPlaneWildcardCert` is false, the ingress is destroyed.

Reply:
ok, here they are